cli/011-cli-multi-config.md

```diff
@@ -0,0 +1,74 @@
+- Start Date: 2021-08-22
+- RFC PR:
+- Issue:
+
+### Configuring multiple environments
+
+##### Problem
+
+Users may have multiple Temporal service environments (development, production, ..) with their own respective mTLS configs, namespaces, address etc. Current implementation of tctl config feature only supports configuring a single environment and doesn't help at quickly switching between multiple.
+
+##### Proposal
+
+To make it easy for users to switch between environments, allow configuring multiple instead of just one and provide command to switch between them.
+
+Syntax to activate environment configs:
+
+```bash
+$ tctl config use-environment <name>
+Active environment: <name>
+```
+
+If the environment doesn't yet exist in config, create it and set as active
+
+```bash
+$ tctl config use-environment <new>
+Environment <new> record doesn't exist, adding..
+Active environment: <name>
+```
+
+Split configuration keys into 1) environment scope - keys residing in a scope of specific environments 2) global scope - out of any specific environment
+
+Global scope keys:
+
+- "active" - name of active environment
+- "environments" - sequence of environments
+- "alias" - sequence of command aliases
+
+Environment scope keys:
```

Comment:
What's the default environment? How do I use this with a single temporal env?

Reply:
thinking of `local` as the default env

Reply:
Is a "default" environment needed? I think if there is no active environment present then `tctl` default flag values should apply. To override defaults for a single temporal env without flags for subsequent commands, you can create a single environment and set it as active.

Maybe a special `default` environment could automatically be created though if someone attempts to set a config value without an active environment set?

Reply:
I'm thinking about the scenario where a user adds the first env.
Does it automatically get set as `active`? If so how do users connect to their "local" cluster again? Do they have to explicitly create another `local` env?
Having a `local` env in the config could be convenient and as a bonus it's now customizable.

So I propose this DX:
```shellsession
$ tctl create-env new-env --address remote-cluster:7223
Environment `new-env` created, available environments:
- local (CURRENT)
- new-env
To switch to environments use `tctl use <ENV_NAME>
```